### ISSUE TITLE
[Documentation request] Batching clarification

### ISSUE THREAD

**Issue by austinmw** [webdataset/webdataset#159]
I'm very confused about the new readme "complete pipeline" example.

Why does it do `dataset.batched(16)`, then `wds.WebLoader(..., batch_size=8)`, then `unbatched()`, then `.batched(12)`? This is very confusing and I can't seem to follow.

**Comment by dxli94** [comment]
> Hi, I'm confused about the new readme "complete pipeline" example.
> 
> Why does it do `dataset.batched(16)`, then `wds.WebLoader(..., batch_size=8)`, then `.unbatched()`, then `.batched(12)`? It says, "batch in the dataset and then rebatch after the loader." Why is it that we should batch, then unbatch, then batch again? I think this must shuffle better, but not sure how? And then how do you pick the initial batch size?
> 
> Also for multinode batching, should we still initialize dataloader like this when using `ResampledShards`?
> 
> ```python
> loader.length = dataset_size // batch_size
> loader = loader.ddp_equalize(dataset_size // batch_size)
> ```
> 
> And finally, I noticed that total epochs seems to be `loader.num_workers * .with_epoch(N)`. So if for example, we have a dataset of 60k samples, batch size of 1000, 2 nodes, each with 8 gpus, and 8 workers per gpu, how would we set `with_epoch()` so that we train on each image roughly once per epoch?
> 
> Thanks!

I have the exact same set of question in my mind. Would appreciate for more clarification.
